### ISSUE TITLE
Use @Block annotation to declare blocks. 

### ISSUE THREAD

**Issue by drupol** [fabianx-ai/service_container#68]
I had a bit of fun this morning and did this. It can be improved a lot, but it works already. I just post this also as a reminder for me.

File: service_container/modules/service_container_doctrine/service_container_doctrine.module

``` php
<?php

include DRUPAL_ROOT . '/' . drupal_get_path('module', 'service_container') . '/lib/Drupal/Component/Annotation/Plugin/Discovery/AnnotatedClassDiscovery.php';

use Drupal\Component\Annotation\Plugin\Discovery\AnnotatedClassDiscovery;

/**
 * Implements hook_block_info().
 */
function service_container_doctrine_block_info() {
  $blocks = array();
  $definitions = _get_blocks_definitions();

  foreach($definitions as $definition) {
    $blocks[$definition['id']]['info'] = $definition['admin_label'];
  }

  return $blocks;
}

/**
 * Implements hook_block_view().
 */
function service_container_doctrine_block_view($delta = '') {
  $definition = _get_blocks_definitions($delta);

  if ($definition) {
    $classname = 'Drupal\\' . $definition['class'];
    $block = new $classname;
    return array(
      'content' => $block->build()
    );
  }
}

/**
 * Helper function, returns Blocks definitions.
 */
function _get_blocks_definitions($delta = '') {
  $namespaces = array();
  foreach (module_list() as $module_name) {
    $namespaces[$module_name] = array(DRUPAL_ROOT . '/' . drupal_get_path('module', $module_name) . '/src');
  }

  $discovery = new AnnotatedClassDiscovery($namespaces, 'Drupal\Core\Block\Annotation\Block');
  $definitions = $discovery->getDefinitions();

  if (!empty($delta) && isset($definitions[$delta])) {
    return $definitions[$delta];
  }

  if (empty($delta)) {
    return $definitions;
  }

  return FALSE;
}
```

File: service_container/src/Plugin/Block/TestBlock.php

``` php
<?php

/**
 * @file
 * Contains \Drupal\service_container\Plugin\Block\TestBlock.
 */

namespace Drupal\service_container\Plugin\Block;

/**
 * Defines a Drupal Block.
 *
 * @Block(
 *   id = "TestBlock",
 *   admin_label = "TestBlock admin label",
 *   label = "TestBlock",
 *   category = "Novelty"
 * )
 */
class TestBlock {
  /**
   * {@inheritdoc}
   */
  public function build() {
    return 'Hello World !';
  }
}
```

**Comment by drupol** [comment]
Fully working example committed at: https://github.com/drupol/service_container/commit/aff03f4bd96ce0145ec0520345edf7c36120f61f

**Comment by drupol** [comment]
Merged #66
